src/net/DNS/DNSManager.cs

```diff
@@ -48,6 +54,62 @@ public LookupRequest(string hostname, QType queryType, int timeout, List<IPEndPo
                 DNSServers = dnsServers;
                 CompleteEvent = completeEvent;
             }
+
+            public string id()
+            {
+                return string.Format("{0}{1}", this.QueryType.ToString(), this.Hostname);
+            }
+            #region IDisposable Support
+            private bool disposedValue = false; // To detect redundant calls
+
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!disposedValue)
+                {
+                    if (disposing)
+                    {
+                        // dispose managed state (managed objects).
+                        if (this.CompleteEvent != null)
+                        {
+                            this.CompleteEvent.Set();
+                            this.CompleteEvent.Dispose();
+                            this.CompleteEvent = null;
+                        }
+                        if (this.DNSServers != null)
+                        {
+                            this.DNSServers.Clear();
+                            this.DNSServers = null;
+                        }
+                        if (this.Duplicates != null)
+                        {
+                            this.Duplicates.Clear();
+                            this.Duplicates = null;
+                        }
+                    }
+
+                    // free unmanaged resources (unmanaged objects) and override a finalizer below.
+                    // set large fields to null.
+
+                    disposedValue = true;
+                }
+            }
+
+            // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
+            ~LookupRequest()

```

Comment:
Are we using any unmanaged resources here that we have to dispose of?

Reply:
There are no unmanaged objects, but this is the default Interface implementation of IDisposable.